compose/ui/ui/src/uikitMain/kotlin/androidx/compose/ui/interop/UIKitInteropView.uikit.kt

```diff
@@ -0,0 +1,193 @@
+ */
+@Composable
+fun <T : UIView> UIKitInteropView(
+    modifier: Modifier,
+    background: Color = Color.White,
+    update: (T) -> Unit = NoOpUpdate,
+    dispose: (T) -> Unit = {},
+    factory: () -> T,
+) {
+    val componentInfo = remember { ComponentInfo<T>() }
+    val root = LocalLayerContainer.current
+    val density = LocalDensity.current.density
+    var rectInPixels by remember { mutableStateOf(IntRect(0, 0, 0, 0)) }
+    var localToWindowOffset: IntOffset by remember { mutableStateOf(IntOffset.Zero) }
+    Place(
+        modifier.onGloballyPositioned { childCoordinates ->
+            val coordinates = childCoordinates.parentCoordinates!!
+            localToWindowOffset = coordinates.localToWindow(Offset.Zero).round()
+            val newRectInPixels = IntRect(localToWindowOffset, coordinates.size)
+            if (rectInPixels != newRectInPixels) {
+                val rect = newRectInPixels / density
+                componentInfo.container.setFrame(rect.toCGRect())
+                if (rectInPixels.width != newRectInPixels.width || rectInPixels.height != newRectInPixels.height) {
+                    componentInfo.component.setFrame(
+                        CGRectMake(
+                            0.0,
+                            0.0,
+                            rect.width.toDouble(),
+                            rect.height.toDouble()
+                        )
+                    )
+                }
+                rectInPixels = newRectInPixels
+            }
+        }.drawBehind {
+            drawRect(Color.Transparent, blendMode = BlendMode.DstAtop)//draw transparent hole
+        }
+    )
+
+    DisposableEffect(factory) {
+        componentInfo.component = factory()
+        componentInfo.container = UIView().apply {
+            addSubview(componentInfo.component)
+        }
+        componentInfo.updater = Updater(componentInfo.component, update)
+        root.insertSubview(componentInfo.container, 0)
+        onDispose {
+            componentInfo.container.removeFromSuperview()
+            componentInfo.updater.dispose()
+            dispose(componentInfo.component)
+        }
+    }
+    SideEffect {
+        componentInfo.container.backgroundColor = parseColor(background)
+        componentInfo.updater.update = update
+    }
+}
+
+@Composable
+private fun Place(modifier: Modifier) {
+    Layout({}, measurePolicy = PlaceMeasurePolicy, modifier = modifier)
+}
+
+private object PlaceMeasurePolicy : MeasurePolicy {
+    override fun MeasureScope.measure(measurables: List<Measurable>, constraints: Constraints) =
+        layout(constraints.maxWidth, constraints.maxHeight) {}
+}
+
+private fun parseColor(color: Color): UIColor {
+    return UIColor(
+        red = color.red.toDouble(),
+        green = color.green.toDouble(),
+        blue = color.blue.toDouble(),
+        alpha = color.alpha.toDouble()
+    )
+}
+
+private class ComponentInfo<T : UIView> {
+    lateinit var container: UIView
```

Comment:
Do we need container?

Reply:
I checked the container helps with MKMapView interop.
We move container instead of view and MKMapView not rendered in this case.